errors/errors.go

```diff
@@ -101,13 +102,14 @@ func (e Error) ABCICode() uint32 { return e.code }
 // this error. Below two lines are equal
 //   e.New("my description")
 //   Wrap(e, "my description")
-func (e Error) New(description string) error {
-	return Wrap(e, description)
+// Allows sprintf format and vararg
+func (e Error) New(description string, args ...interface{}) error {
+	return Wrap(e, fmt.Sprintf(description, args...))
```

Comment:
Ha! This was my request in another pr.

Please *do not* do this in New.... we should avoid Sprintf when not needed.
But I *do* like the idea of this function you just wrote called `Newf`, like `fmt.Errorf`

Just keep the old New, and rename this function and we are all happy :smile: